src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -266,7 +266,7 @@ public function save()
             session_write_close();
         } finally {
             restore_error_handler();
-            $_SESSION = $session;
+            $_SESSION = $session; // TODO: why is $_SESSION restored with the meta data and bag keys here?
```

Comment:
I think I know why this is. It's to restore the bags in case there were other keys. That's strange though.
Let me add to this PR on what I think would be correct. Maybe that's easier for you to review then when you can actually see the changes.